moveit_core/online_signal_smoothing/test/test_acceleration_filter.cpp

```diff
@@ -0,0 +1,196 @@
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#include <gtest/gtest.h>
+#include <moveit/online_signal_smoothing/acceleration_filter.h>
+#include <moveit/utils/robot_model_test_utils.h>
+
+class AccelerationFilterTest : public testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    robot_model_ = moveit::core::loadTestingRobotModel("panda");
+  };
+
+  void set_limit(std::optional<Eigen::VectorXd> acceleration_limit)
+  {
+    const std::vector<moveit::core::JointModel*> joint_models = robot_model_->getJointModels();
+    auto joint_model_group = robot_model_->getJointModelGroup(move_group_name_);
+    size_t ind = 0;
+    for (auto& joint_model : joint_models)
+    {
+      if (!joint_model_group->hasJointModel(joint_model->getName()))
+      {
+        continue;
+      }
+      std::vector<moveit_msgs::msg::JointLimits> joint_bounds_msg(joint_model->getVariableBoundsMsg());
+      for (auto& joint_bound : joint_bounds_msg)
+      {
+        joint_bound.has_acceleration_limits = acceleration_limit.has_value();
+        if (joint_bound.has_acceleration_limits)
+        {
+          joint_bound.max_acceleration = acceleration_limit.value()[ind];
+        }
+      }
+      joint_model->setVariableBounds(joint_bounds_msg);
+      ind++;
+    }
+  }
+
+protected:
+  moveit::core::RobotModelPtr robot_model_;
+  std::string move_group_name_ = "panda_arm";
+};
+
+TEST_F(AccelerationFilterTest, FilterInitialize)
+{
+  online_signal_smoothing::AccelerationLimitedPlugin filter;
+
+  rclcpp::Node::SharedPtr node = std::make_shared<rclcpp::Node>("AccelerationFilterTest");
+  node->declare_parameter<std::string>("move_group_name", move_group_name_);
+
+  // fail because the number of joint is wrong
+  EXPECT_FALSE(filter.initialize(node, robot_model_, 3));
+
+  // fail because the update_rate parameter is not set
+  EXPECT_FALSE(filter.initialize(node, robot_model_, 7));
```

Comment:
The `7` (or `7u`) is used in LOTS of lines throughout. Maybe consider also defining a `constexpr` up top to make life easier:

```cpp
constexpr size_t PANDA_NUM_JOINTS = 7u
```

(I think the moveit2 repo uses this standard for constant naming, but clang_tidy will yell at you otherwise)

Or if the caps look TOO COOL, maybe just define the const variables at the beginning of each test, or as a member variable of the fixture.

Reply:
Okay, I moved it to the top as a constexpr